pkg/upload/js.go

```diff
@@ -108,6 +109,101 @@ func resolveSourceMapPaths(sourceMapPath string, outputPath string, logger log.L
 	return sourceMapPaths, nil
 }
 
+// Reads a JSON source map into memory. Required in order to check if the sourcesContent exists.
+func ReadSourceMap(path string, logger log.Logger) (map[string]interface{}, error) {
+	logger.Debug(fmt.Sprintf("Reading sourcemap %s", path))
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return map[string]interface{}{}, err
+	}
+	var sourceMapContents map[string]interface{}
+	err = json.Unmarshal(data, &sourceMapContents)
+	if err != nil {
+		return map[string]interface{}{}, err
+	}
+	return sourceMapContents, nil
+}
+
+func addSourcesContent(section map[string]interface{}, relativePath string, logger log.Logger) {
```

Comment:
I previously said we shoud reserve that term for the string we pass to the API only. However it seems justified to call it `projectRoot` here (and in calling function) as this is specifically the project root we want (and not necessarily the base directory of the project, if these are different)

```suggestion
func addSourcesContent(section map[string]interface{}, projectRoot string, logger log.Logger) {
```